i6_models/config.py

```diff
@@ -51,3 +53,22 @@ def _validate_types(self) -> None:
 
     def __post_init__(self) -> None:
         self._validate_types()
+
+
+ConfigType = TypeVar("ConfigType", bound=ModelConfiguration)
+ModuleType = TypeVar("ModuleType", bound=nn.Module)
```

Comment:
We might need to extend `ModuleType` even further, as a `Protocol` or so, to define that we expect an `__init__` with one single argument `cfg: ConfigType`.

Reply:
I think this should be a follow up PR too.

Reply:
If you do not fix this here, I think you might see a warning in PyCharm, mypy and other tools about wrong typing somewhere. Not sure.

Reply:
With this the example I posted above does not recieve a PyTorch warning.

Reply:
I mean in the `construct` method. Or maybe in other places. Not for the creation of the cfg/opts object.